src/rime_api.cc

```diff
@@ -1076,13 +1076,29 @@ Bool RimeSelectCandidateOnCurrentPage(RimeSessionId session_id, size_t index) {
     return False;
   const auto& seg(ctx->composition().back());
   size_t page_start = seg.selected_index / page_size * page_size;
-  return Bool(ctx->Select(page_start + index));
+  return (ctx->*verb)(page_start + index);
+}
+
+Bool RimeSelectCandidate(RimeSessionId session_id, size_t index) {
+  return do_with_candidate(session_id, index, Select);
```

Comment:
Prefer explicit conversion on the returned C type (`Bool`=`int`):
`return Bool(do_with_candidate(...));`

Prefer `&Select` to make it obvious we are passing a function pointer.